File: js/mgtTroll.js
var nm1 = ["b","bh","br","d","dh","dr","g","gh","gr","kh","khr","r","th","tr","thr","v","z","zh"];
var nm2 = ["u","u","u","u","u","o","a"];
var nm3 = ["b","bv","bb","bl","d","dd","dr","dv","dz","g","gg","gl","gn","gv","gz","k","kk","kl","kv","lb","ld","lg","l","ll","ln","lr","lt","lz","lv","m","md","mg","ml","mr","mz","nd","nv","nz","nl","nd","ndr","ng","ngr","ngb","ngz","r","rr","rl","rv","rz","v","vl","vn","z","zd","zg","zl","zn"];
var nm4 = ["c","d","g","l","lg","ld","n","ng","nd","rd","rg"];

var nm5 = ["amber","ash","bitter","blaze","blunt","bone","boulder","brick","bristle","broad","cave","cinder","claw","crag","dim","dirge","dust","far","fern","flask","flint","fog","fore","forge","gloom","grand","gravel","grim","heavy","hill","horn","keen","krag","lone","low","moss","mourn","mud","oat","orb","pale","pyre","rend","saur","shadow","skull","sky","slate","snake","spore","stone","tall","terra","tree","tusk","whit","wild"];
var nm6 = ["back","bane","beam","belly","belt","bend","blade","blight","bough","braid","branch","brand","breath","brew","bridge","brow","claw","crag","crest","flare","flaw","force","fury","gaze","grip","grog","gut","hide","hood","horn","howl","husk","jaw","lance","lash","limb","lock","mane","mark","maw","scar","scrub","shard","shroud","snarl","spine","stride","strike","stub","surge","toe","trap"];
var nm7 = ["Aggressor","Assailant","Barbarian","Battler","Brawler","Bruiser","Brute","Bully","Castaway","Degenerate","Derelict","Exile","Fiend","Hireling","Merc","Mercenary","Outcast","Sadist","Savage","Scrapper","Shaman","Slugger","Troll","Troll","Troll","Troll","Troll","Troll","Troll","Troll","Troll","Troll","Troll","Troll","Troll","Trow","Trow","Trow","Trow","Trow","Trow","Vagabond","Vagrant","Warrior","Wildcat","Wildling","Wretch"];

var nm8 = ["Abandoned","Aching","Aged","Aggressive","Albino","Ancient","Angry","Anxious","Arctic","Bitter","Blind","Bony","Broken","Bruised","Bush","Careless","Cave","Cavern","Charging","Clever","Clueless","Clumsy","Colossal","Confused","Corrupt","Corrupted","Crafty","Crooked","Cruel","Defiant","Diligent","Dull","Eager","Elder","Enraged","Fearless","Feisty","Fickle","Forest","Forsaken","Fungus","Gloomy","Grave","Greedy","Grim","Grotesque","Harvester","Haunting","Hedge","Hidden","Horned","Hungry","Hunting","Jaded","Jungle","Juvenile","Lanky","Limping","Lone","Lost","Lumbering","Marsh","Meager","Mountain","Noxious","Numb","Oblivious","Ocean","Powerful","Primal","Prime","Pygmy","Rash","Reckless","River","Sea","Selfish","Shady","Shameless","Skinny","Sneaking","Sneaky","Sniveling","Swamp","Swift","Troubled","Twin","Vicious","Vile","Wicked","Wrathful","Wretched"];

var br = "";

function nameGen(){
	$('#placeholder').css('textTransform', 'capitalize');
	var element = document.createElement("div");
	element.setAttribute("id", "result");

	for(i = 0; i < 10; i++){
		if(i < 6){
			rnd = Math.random() * nm1.length | 0;
			rnd2 = Math.random() * nm2.length | 0;
			rnd3 = Math.random() * nm4.length | 0;
			while(nm4[rnd3] === nm1[rnd]){
				rnd3 = Math.random() * nm4.length | 0;
			}
			if(i < 3){
				names = nm1[rnd] + nm2[rnd2] + nm4[rnd3];
			}else{
				rnd4 = Math.random() * nm3.length | 0;
				rnd5 = Math.random() * nm2.length | 0;
				while(nm4[rnd3] === nm3[rnd4]){
					rnd4 = Math.random() * nm3.length | 0;
				}
				names = nm1[rnd] + nm2[rnd2] + nm3[rnd4] + nm2[rnd5] + nm4[rnd3];
			}
		}else if(i < 8){
			rnd = Math.random() * nm5.length | 0;
			rnd2 = Math.random() * nm6.length | 0;
			while(nm5[rnd] === nm6[rnd2]){
				rnd2 = Math.random() * nm6.length | 0;
			}
			rnd3 = Math.random() * nm7.length | 0;
			names = nm5[rnd] + nm6[rnd2] + " " + nm7[rnd3];
		}else{
			rnd = Math.random() * nm8.length | 0;
			rnd2 = Math.random() * nm7.length | 0;
			names = nm8[rnd] + " " + nm7[rnd2];
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}

File: generators/mgtTroll.py
__all__ = ['mgtTroll']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm4', 'nameGen', 'nm5', 'nm8', 'nm3', 'nm7', 'nm2', 'nm6'])
@Js
def PyJsHoisted_nameGen_(this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this}, var)
    var.registers(['result'])
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            if (var.get('i')<Js(6.0)):
                var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm1').get('length'))|Js(0.0)))
                var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm2').get('length'))|Js(0.0)))
                var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm4').get('length'))|Js(0.0)))
                while PyJsStrictEq(var.get('nm4').get(var.get('rnd3')),var.get('nm1').get(var.get('rnd'))):
                    var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm4').get('length'))|Js(0.0)))
                if (var.get('i')<Js(3.0)):
                    var.put('names', ((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm4').get(var.get('rnd3'))))
                else:
                    var.put('rnd4', ((var.get('Math').callprop('random')*var.get('nm3').get('length'))|Js(0.0)))
                    var.put('rnd5', ((var.get('Math').callprop('random')*var.get('nm2').get('length'))|Js(0.0)))
                    while PyJsStrictEq(var.get('nm4').get(var.get('rnd3')),var.get('nm3').get(var.get('rnd4'))):
                        var.put('rnd4', ((var.get('Math').callprop('random')*var.get('nm3').get('length'))|Js(0.0)))
                    var.put('names', ((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd4')))+var.get('nm2').get(var.get('rnd5')))+var.get('nm4').get(var.get('rnd3'))))
            else:
                if (var.get('i')<Js(8.0)):
                    var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm5').get('length'))|Js(0.0)))
                    var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm6').get('length'))|Js(0.0)))
                    while PyJsStrictEq(var.get('nm5').get(var.get('rnd')),var.get('nm6').get(var.get('rnd2'))):
                        var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm6').get('length'))|Js(0.0)))
                    var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm7').get('length'))|Js(0.0)))
                    var.put('names', (((var.get('nm5').get(var.get('rnd'))+var.get('nm6').get(var.get('rnd2')))+Js(' '))+var.get('nm7').get(var.get('rnd3'))))
                else:
                    var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm8').get('length'))|Js(0.0)))
                    var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm7').get('length'))|Js(0.0)))
                    var.put('names', ((var.get('nm8').get(var.get('rnd'))+Js(' '))+var.get('nm7').get(var.get('rnd2'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js('b'), Js('bh'), Js('br'), Js('d'), Js('dh'), Js('dr'), Js('g'), Js('gh'), Js('gr'), Js('kh'), Js('khr'), Js('r'), Js('th'), Js('tr'), Js('thr'), Js('v'), Js('z'), Js('zh')]))
var.put('nm2', Js([Js('u'), Js('u'), Js('u'), Js('u'), Js('u'), Js('o'), Js('a')]))
var.put('nm3', Js([Js('b'), Js('bv'), Js('bb'), Js('bl'), Js('d'), Js('dd'), Js('dr'), Js('dv'), Js('dz'), Js('g'), Js('gg'), Js('gl'), Js('gn'), Js('gv'), Js('gz'), Js('k'), Js('kk'), Js('kl'), Js('kv'), Js('lb'), Js('ld'), Js('lg'), Js('l'), Js('ll'), Js('ln'), Js('lr'), Js('lt'), Js('lz'), Js('lv'), Js('m'), Js('md'), Js('mg'), Js('ml'), Js('mr'), Js('mz'), Js('nd'), Js('nv'), Js('nz'), Js('nl'), Js('nd'), Js('ndr'), Js('ng'), Js('ngr'), Js('ngb'), Js('ngz'), Js('r'), Js('rr'), Js('rl'), Js('rv'), Js('rz'), Js('v'), Js('vl'), Js('vn'), Js('z'), Js('zd'), Js('zg'), Js('zl'), Js('zn')]))
var.put('nm4', Js([Js('c'), Js('d'), Js('g'), Js('l'), Js('lg'), Js('ld'), Js('n'), Js('ng'), Js('nd'), Js('rd'), Js('rg')]))
var.put('nm5', Js([Js('amber'), Js('ash'), Js('bitter'), Js('blaze'), Js('blunt'), Js('bone'), Js('boulder'), Js('brick'), Js('bristle'), Js('broad'), Js('cave'), Js('cinder'), Js('claw'), Js('crag'), Js('dim'), Js('dirge'), Js('dust'), Js('far'), Js('fern'), Js('flask'), Js('flint'), Js('fog'), Js('fore'), Js('forge'), Js('gloom'), Js('grand'), Js('gravel'), Js('grim'), Js('heavy'), Js('hill'), Js('horn'), Js('keen'), Js('krag'), Js('lone'), Js('low'), Js('moss'), Js('mourn'), Js('mud'), Js('oat'), Js('orb'), Js('pale'), Js('pyre'), Js('rend'), Js('saur'), Js('shadow'), Js('skull'), Js('sky'), Js('slate'), Js('snake'), Js('spore'), Js('stone'), Js('tall'), Js('terra'), Js('tree'), Js('tusk'), Js('whit'), Js('wild')]))
var.put('nm6', Js([Js('back'), Js('bane'), Js('beam'), Js('belly'), Js('belt'), Js('bend'), Js('blade'), Js('blight'), Js('bough'), Js('braid'), Js('branch'), Js('brand'), Js('breath'), Js('brew'), Js('bridge'), Js('brow'), Js('claw'), Js('crag'), Js('crest'), Js('flare'), Js('flaw'), Js('force'), Js('fury'), Js('gaze'), Js('grip'), Js('grog'), Js('gut'), Js('hide'), Js('hood'), Js('horn'), Js('howl'), Js('husk'), Js('jaw'), Js('lance'), Js('lash'), Js('limb'), Js('lock'), Js('mane'), Js('mark'), Js('maw'), Js('scar'), Js('scrub'), Js('shard'), Js('shroud'), Js('snarl'), Js('spine'), Js('stride'), Js('strike'), Js('stub'), Js('surge'), Js('toe'), Js('trap')]))
var.put('nm7', Js([Js('Aggressor'), Js('Assailant'), Js('Barbarian'), Js('Battler'), Js('Brawler'), Js('Bruiser'), Js('Brute'), Js('Bully'), Js('Castaway'), Js('Degenerate'), Js('Derelict'), Js('Exile'), Js('Fiend'), Js('Hireling'), Js('Merc'), Js('Mercenary'), Js('Outcast'), Js('Sadist'), Js('Savage'), Js('Scrapper'), Js('Shaman'), Js('Slugger'), Js('Troll'), Js('Troll'), Js('Troll'), Js('Troll'), Js('Troll'), Js('Troll'), Js('Troll'), Js('Troll'), Js('Troll'), Js('Troll'), Js('Troll'), Js('Troll'), Js('Troll'), Js('Trow'), Js('Trow'), Js('Trow'), Js('Trow'), Js('Trow'), Js('Trow'), Js('Vagabond'), Js('Vagrant'), Js('Warrior'), Js('Wildcat'), Js('Wildling'), Js('Wretch')]))
var.put('nm8', Js([Js('Abandoned'), Js('Aching'), Js('Aged'), Js('Aggressive'), Js('Albino'), Js('Ancient'), Js('Angry'), Js('Anxious'), Js('Arctic'), Js('Bitter'), Js('Blind'), Js('Bony'), Js('Broken'), Js('Bruised'), Js('Bush'), Js('Careless'), Js('Cave'), Js('Cavern'), Js('Charging'), Js('Clever'), Js('Clueless'), Js('Clumsy'), Js('Colossal'), Js('Confused'), Js('Corrupt'), Js('Corrupted'), Js('Crafty'), Js('Crooked'), Js('Cruel'), Js('Defiant'), Js('Diligent'), Js('Dull'), Js('Eager'), Js('Elder'), Js('Enraged'), Js('Fearless'), Js('Feisty'), Js('Fickle'), Js('Forest'), Js('Forsaken'), Js('Fungus'), Js('Gloomy'), Js('Grave'), Js('Greedy'), Js('Grim'), Js('Grotesque'), Js('Harvester'), Js('Haunting'), Js('Hedge'), Js('Hidden'), Js('Horned'), Js('Hungry'), Js('Hunting'), Js('Jaded'), Js('Jungle'), Js('Juvenile'), Js('Lanky'), Js('Limping'), Js('Lone'), Js('Lost'), Js('Lumbering'), Js('Marsh'), Js('Meager'), Js('Mountain'), Js('Noxious'), Js('Numb'), Js('Oblivious'), Js('Ocean'), Js('Powerful'), Js('Primal'), Js('Prime'), Js('Pygmy'), Js('Rash'), Js('Reckless'), Js('River'), Js('Sea'), Js('Selfish'), Js('Shady'), Js('Shameless'), Js('Skinny'), Js('Sneaking'), Js('Sneaky'), Js('Sniveling'), Js('Swamp'), Js('Swift'), Js('Troubled'), Js('Twin'), Js('Vicious'), Js('Vile'), Js('Wicked'), Js('Wrathful'), Js('Wretched')]))
pass
pass


# Add lib to the module scope
mgtTroll = var.to_python()